### PR TITLE
Simplify rollback visitor for send blocks

### PR DESCRIPTION
**Pull request by cryptocode**

### CODE REVIEW

File: rai/ledger.cpp [hunk 1]
```diff
@@ -144,18 +144,9 @@ class rollback_visitor : public rai::block_visitor
 		if (previous != nullptr)
 		{
 			ledger.store.block_successor_clear (transaction, block_a.hashables.previous);
-			switch (previous->type ())
+			if (previous->type () != rai::block_type::state)
```
[the remark this answers is not in the document]
Comment (cryptocode) [reply rai/ledger.cpp]: I personally find != clearer as the reader doesn't need to know about the enum order, but that sounds close to bikeshedding ;) I assume the code would need to change anyway if more block types are added. I can change it if you want, lemme know.

[the remark this answers is not in the document]
Comment (PlasmaPower) [reply rai/ledger.cpp]: I'm just thinking that future block types will probably be improvements on state blocks. With the enum, new block types will not use frontier_put. With this if statement, new blocks will use frontier_put. I think the former is the correct behavior.

[the remark this answers is not in the document]
Comment (cryptocode) [reply rai/ledger.cpp]: Changed to <

**Comment by PlasmaPower** [comment]
LGTM

**Comment by clemahieu** [comment]
I picked the switch statement so if we add new block types the compiler will error that its missing an enumeration and we'll be required to come to this code section and evaluate what should be done.

Not sure if it's really needed though.

**Comment by cryptocode** [comment]
You wouldn't get an error because there's a default branch.

**Comment by clemahieu** [comment]
Ahh right.  Looks good.

**Pull request merged**